### PR TITLE
Provide useful tips for those who explores GraphQL schema interactively

### PR DESCRIPTION
**Pull request by TomashKhamlai**
### Description (*)

Provide useful tips for those who explore GraphQL schema interactively.

### Fixed Issues (if relevant)

1. magento/graphql-ce#655: Add/Update docblock descriptions in GraphQL schema for modules related to Quote operations

### Manual testing scenarios (*)

1. Check that now it is clear which types were deprecated in autocomplete dropdown in GraphQL Playground

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/ConfigurableProductGraphQl/etc/schema.graphqls [hunk 1]
```diff
@@ -50,12 +50,12 @@ type AddConfigurableProductsToCartOutput {
...
 input ConfigurableProductCartItemInput {
     data: CartItemInput!
-    variant_sku: String @deprecated(reason: "Use CartItemInput.sku instead.")
+    variant_sku: String @deprecated(reason: "Use CartItemInput.sku instead.") @doc(description: "Deprecated: use CartItemInput.sku instead.")
     parent_sku: String @doc(description: "Configurable product SKU.")
     customizable_options:[CustomizableOptionInput!]
 }
 
-type ConfigurableCartItem implements CartItemInterface {
+type ConfigurableCartItem implements CartItemInterface @doc(description: "Provides data specific to Cart Items with Configurable Product.") {
```
Comment (lenaorobei) [inline app/code/Magento/ConfigurableProductGraphQl/etc/schema.graphqls]:
```suggestion
type ConfigurableCartItem implements CartItemInterface @doc(description: "Provides cart item options for configurable product.") {
```

File: app/code/Magento/DownloadableGraphQl/etc/schema.graphqls [hunk 2]
```diff
@@ -28,7 +28,7 @@ type AddDownloadableProductsToCartOutput {
     cart: Cart!
 }
 
-type DownloadableCartItem implements CartItemInterface @doc(description: "Downloadable Cart Item") {
+type DownloadableCartItem implements CartItemInterface @doc(description: "Provides data specific to Cart Items with Downloadable Product.") {
```
Comment (lenaorobei) [inline app/code/Magento/DownloadableGraphQl/etc/schema.graphqls]:
```suggestion
type DownloadableCartItem implements CartItemInterface @doc(description: "Provides cart item options for downloadable product.") {
```

File: app/code/Magento/DownloadableGraphQl/etc/schema.graphqls [hunk 3]
```diff
@@ -59,23 +59,23 @@ enum DownloadableFileTypeEnum @deprecated(reason: "`sample_url` serves to get th
 }
 
 type DownloadableProductLinks @doc(description: "DownloadableProductLinks defines characteristics of a downloadable product") {
-    id: Int @deprecated(reason: "This information shoud not be exposed on frontend")
+    id: Int @deprecated(reason: "This information shoud not be exposed on frontend") @doc(description: "Deprecated: will return null")
```
Comment (lenaorobei) [inline app/code/Magento/DownloadableGraphQl/etc/schema.graphqls]:
```suggestion
    id: Int @deprecated(reason: "This information should not be exposed on frontend") @doc(description: "Deprecated: for security reasons will return null.")
```

File: app/code/Magento/DownloadableGraphQl/etc/schema.graphqls [hunk 4]
```diff
@@ -59,23 +59,23 @@ enum DownloadableFileTypeEnum @deprecated(reason: "`sample_url` serves to get th
...
     title: String @doc(description: "The display name of the link")
     sort_order: Int @doc(description: "A number indicating the sort order")
     price: Float @doc(description: "The price of the downloadable product")
     sample_url: String @doc(description: "URL to the downloadable sample")
-    is_shareable: Boolean @deprecated(reason: "This information shoud not be exposed on frontend")
-    number_of_downloads: Int @deprecated(reason: "This information shoud not be exposed on frontend")
-    link_type: DownloadableFileTypeEnum @deprecated(reason: "`sample_url` serves to get the downloadable sample")
-    sample_type: DownloadableFileTypeEnum @deprecated(reason: "`sample_url` serves to get the downloadable sample")
-    sample_file: String @deprecated(reason: "`sample_url` serves to get the downloadable sample")
+    is_shareable: Boolean @deprecated(reason: "This information shoud not be exposed on frontend") @doc(description: "Deprecated: will return null")
```
Comment (lenaorobei) [inline app/code/Magento/DownloadableGraphQl/etc/schema.graphqls]:
```suggestion
    is_shareable: Boolean @deprecated(reason: "This information should not be exposed on frontend") @doc(description: "Deprecated: for security reasons will return null.")
```

File: app/code/Magento/DownloadableGraphQl/etc/schema.graphqls [hunk 5]
```diff
@@ -59,23 +59,23 @@ enum DownloadableFileTypeEnum @deprecated(reason: "`sample_url` serves to get th
...
+    number_of_downloads: Int @deprecated(reason: "This information shoud not be exposed on frontend") @doc(description: "Deprecated: will return null")
```
Comment (lenaorobei) [inline app/code/Magento/DownloadableGraphQl/etc/schema.graphqls]:
```suggestion
    number_of_downloads: Int @deprecated(reason: "This information should not be exposed on frontend") @doc(description: "Deprecated: for security reasons will return null.")
```

File: app/code/Magento/DownloadableGraphQl/etc/schema.graphqls [hunk 6]
```diff
@@ -59,23 +59,23 @@ enum DownloadableFileTypeEnum @deprecated(reason: "`sample_url` serves to get th
...
+    link_type: DownloadableFileTypeEnum @deprecated(reason: "`sample_url` serves to get the downloadable sample") @doc(description: "Deprecated: will return null")
```
Comment (lenaorobei) [inline app/code/Magento/DownloadableGraphQl/etc/schema.graphqls]:
```suggestion
    link_type: DownloadableFileTypeEnum @deprecated(reason: "`sample_url` serves to get the downloadable sample") @doc(description: "Deprecated: for security reasons will return null.")
```

File: app/code/Magento/DownloadableGraphQl/etc/schema.graphqls [hunk 7]
```diff
@@ -59,23 +59,23 @@ enum DownloadableFileTypeEnum @deprecated(reason: "`sample_url` serves to get th
...
+    sample_type: DownloadableFileTypeEnum @deprecated(reason: "`sample_url` serves to get the downloadable sample") @doc(description: "Deprecated: will return null")
```
Comment (lenaorobei) [inline app/code/Magento/DownloadableGraphQl/etc/schema.graphqls]:
```suggestion
    sample_type: DownloadableFileTypeEnum @deprecated(reason: "`sample_url` serves to get the downloadable sample") @doc(description: "Deprecated: for security reasons will return null.")
```

File: app/code/Magento/DownloadableGraphQl/etc/schema.graphqls [hunk 8]
```diff
@@ -59,23 +59,23 @@ enum DownloadableFileTypeEnum @deprecated(reason: "`sample_url` serves to get th
...
+    sample_file: String @deprecated(reason: "`sample_url` serves to get the downloadable sample") @doc(description: "Deprecated: will return null")
```
Comment (lenaorobei) [inline app/code/Magento/DownloadableGraphQl/etc/schema.graphqls]:
```suggestion
    sample_file: String @deprecated(reason: "`sample_url` serves to get the downloadable sample") @doc(description: "Deprecated: for security reasons will return null.")
```

File: app/code/Magento/DownloadableGraphQl/etc/schema.graphqls [hunk 9]
```diff
@@ -59,23 +59,23 @@ enum DownloadableFileTypeEnum @deprecated(reason: "`sample_url` serves to get th
...
 }
 
 type DownloadableProductSamples @doc(description: "DownloadableProductSamples defines characteristics of a downloadable product") {
-    id: Int @deprecated(reason: "This information shoud not be exposed on frontend")
+    id: Int @deprecated(reason: "This information shoud not be exposed on frontend") @doc(description: "Deprecated: will return null")
```
Comment (lenaorobei) [inline app/code/Magento/DownloadableGraphQl/etc/schema.graphqls]:
```suggestion
    id: Int @deprecated(reason: "This information should not be exposed on frontend") @doc(description: "Deprecated: for security reasons will return null.")
```

File: app/code/Magento/DownloadableGraphQl/etc/schema.graphqls [hunk 10]
```diff
@@ -59,23 +59,23 @@ enum DownloadableFileTypeEnum @deprecated(reason: "`sample_url` serves to get th
...
     title: String @doc(description: "The display name of the sample")
     sort_order: Int @doc(description: "A number indicating the sort order")
     sample_url: String @doc(description: "URL to the downloadable sample")
-    sample_type: DownloadableFileTypeEnum @deprecated(reason: "`sample_url` serves to get the downloadable sample")
-    sample_file: String @deprecated(reason: "`sample_url` serves to get the downloadable sample")
+    sample_type: DownloadableFileTypeEnum @deprecated(reason: "`sample_url` serves to get the downloadable sample") @doc(description: "Deprecated: will return null.")
```
Comment (lenaorobei) [inline app/code/Magento/DownloadableGraphQl/etc/schema.graphqls]:
```suggestion
    sample_type: DownloadableFileTypeEnum @deprecated(reason: "`sample_url` serves to get the downloadable sample") @doc(description: "Deprecated: for security reasons will return null.")
```

File: app/code/Magento/DownloadableGraphQl/etc/schema.graphqls [hunk 11]
```diff
@@ -59,23 +59,23 @@ enum DownloadableFileTypeEnum @deprecated(reason: "`sample_url` serves to get th
...
+    sample_file: String @deprecated(reason: "`sample_url` serves to get the downloadable sample") @doc(description: "Deprecated: will return null.")
```
Comment (lenaorobei) [inline app/code/Magento/DownloadableGraphQl/etc/schema.graphqls]:
```suggestion
    sample_file: String @deprecated(reason: "`sample_url` serves to get the downloadable sample") @doc(description: "Deprecated: for security reasons will return null.")
```

File: app/code/Magento/QuoteGraphQl/etc/schema.graphqls [hunk 12]
```diff
@@ -302,15 +302,15 @@ type SetGuestEmailOnCartOutput {
     cart: Cart!
 }
 
-type SimpleCartItem implements CartItemInterface @doc(description: "Simple Cart Item") {
+type SimpleCartItem implements CartItemInterface @doc(description: "Provides data specific to Cart Items with Simple Product.") {
```
Comment (lenaorobei) [inline app/code/Magento/QuoteGraphQl/etc/schema.graphqls]:
```suggestion
type SimpleCartItem implements CartItemInterface @doc(description: "Provides cart item options for simple product.") {
```

File: app/code/Magento/QuoteGraphQl/etc/schema.graphqls [hunk 13]
```diff
@@ -302,15 +302,15 @@ type SetGuestEmailOnCartOutput {
...
     customizable_options: [SelectedCustomizableOption] @resolver(class: "Magento\\QuoteGraphQl\\Model\\Resolver\\CustomizableOptions")
 }
 
-type VirtualCartItem implements CartItemInterface @doc(description: "Virtual Cart Item") {
+type VirtualCartItem implements CartItemInterface @doc(description: "Provides data specific to Cart Items with Virtual Product.") {
```
Comment (lenaorobei) [inline app/code/Magento/QuoteGraphQl/etc/schema.graphqls]:
```suggestion
type VirtualCartItem implements CartItemInterface @doc(description: "Provides cart item options for virtual product.") {
```

File: app/code/Magento/QuoteGraphQl/etc/schema.graphqls [hunk 14]
```diff
@@ -302,15 +302,15 @@ type SetGuestEmailOnCartOutput {
...
     customizable_options: [SelectedCustomizableOption] @resolver(class: "Magento\\QuoteGraphQl\\Model\\Resolver\\CustomizableOptions")
 }
 
-interface CartItemInterface @typeResolver(class: "Magento\\QuoteGraphQl\\Model\\Resolver\\CartItemTypeResolver") {
+interface CartItemInterface @typeResolver(class: "Magento\\QuoteGraphQl\\Model\\Resolver\\CartItemTypeResolver") @doc(description: "Provides common set of cart item attributes data available for all cart item types.") {
```
Comment (lenaorobei) [inline app/code/Magento/QuoteGraphQl/etc/schema.graphqls]:
```suggestion
interface CartItemInterface @typeResolver(class: "Magento\\QuoteGraphQl\\Model\\Resolver\\CartItemTypeResolver") @doc(description: "Provides common cart item options set suitable for all product types.") {
```

Comment (TomashKhamlai) [reply app/code/Magento/QuoteGraphQl/etc/schema.graphqls]: Done.

**Pull request closed without merging**